sdk/core/Azure.Core/tests/DynamicData/DynamicJsonTests.cs

```diff
@@ -293,6 +294,141 @@ public void CanMakeChangesAndAddNewProperty()
             Assert.AreEqual(2, (int)jsonData.Bar);
         }
 
+        [Test]
+        public void CanAddPocoProperty()
```

Comment:
Without this change, MutableJsonDocument will serialize/deserialize any changes (e.g. calls to Set or SetProperty) with JsonSerializer defaults.  That results in .NET models serializing with PascalCase member names, so the accesses to camelCase member names in this test would fail.  I didn't add tests at the MutableJsonDocument layer, which is probably why it's harder to see.